Spec/Auth.swift

```diff
@@ -3,17 +3,113 @@ import Ably.Private
 import Nimble
 import Quick
 import Aspects
+        
+        private var testHTTPExecutor: TestProxyHTTPExecutor!
```

Comment:
You might be wondering why we pull the variables to global level instead of making them instance variables of the `XCTestCase` subclass. It's because this way, we can access them from inside each `test_` method as if they were local variables, without having to worry about memory management. If they were instance variables, then we’d have to worry about capturing `self` inside closures that refer to these variables.

Reply:
What about `keeping states` by vars? Are these global vars stateless?
We have to be 100% sure that none of the var keeps state that could disrupt any other tests.

This problem isn't probably in the scope of these changes, but we have to re-check it later.

Reply:
Hey @lukasz-szyszkowski, is your question "do these global variables, which are accessed from multiple test cases, mean that we have state which is shared between test cases and which might cause one test case to be affected by another"? If so, the answer is yes. And yes, it's a problem, and it's probably a large part of the reason why the tests are so order-dependent (#1241, #1243). But this is not a new problem – it's the same behaviour as before, where multiple `it` blocks shared the same `spec` / `describe` / `context`- level local variables.

Reply:
Yes, that's what I meant. Is there any Issue with this case?

Reply:
Sorry, but I don't think I understand the question, do you think you could explain a bit more what you'd like to know?